code/modules/projectiles/updated_projectiles/guns/shotguns.dm

```diff
@@ -526,3 +531,95 @@ can cause issues with ammo types getting mixed up during the burst.
 	recoil_unwielded = CONFIG_GET(number/combat_define/high_recoil_value)
 	pump_delay = CONFIG_GET(number/combat_define/mhigh_fire_delay) * 2
 
+//------------------------------------------------------
+//A hacky bolt action rifle. in here for the "pump" or bolt working action.
+
+/obj/item/weapon/gun/shotgun/pump/bolt
+	name = "\improper bolt action rifle"
+	desc = "A hunting rifle just like pappy used to own."
+	icon_state = "mosin"
+	item_state = "mosin" //thank you Alterist
+	fire_sound = 'sound/weapons/gun_sniper.ogg'
+	current_mag = /obj/item/ammo_magazine/internal/shotgun/pump/bolt
+	gun_skill_category = GUN_SKILL_RIFLES
+	type_of_casings = "cartridge"
+	pump_sound = 'sound/weapons/working_the_bolt.ogg'
+	attachable_allowed = list(
+						/obj/item/attachable/reddot,
+						/obj/item/attachable/scope/mini,
+						/obj/item/attachable/scope,
+						/obj/item/attachable/bipod)
+	attachable_offset = list("muzzle_x" = 32, "muzzle_y" = 16,"rail_x" = 8, "rail_y" = 21, "under_x" = 24, "under_y" = 13, "stock_x" = 20, "stock_y" = 14)
+	starting_attachment_types = list(/obj/item/attachable/scope)
+
+/obj/item/weapon/gun/shotgun/pump/bolt/Initialize()
+	. = ..()
+	var/obj/item/attachable/S = new /obj/item/attachable/mosinbarrel(src)
+	S.Attach(src)
+	S = new /obj/item/attachable/stock/mosin(src)
+	S.flags_attach_features &= ~ATTACH_REMOVABLE
```

Comment:
Does this part need to be in the `Initialize`?
I know a lot of items do it, but there's no need to do it on spawn.
Just set the proper flags in the item. Check the parent and remove `ATTACH_REMOVABLE` from it.

Reply:
the SVD does `Initialize` and I copied it from there for length.